File: concrete/src/test/java/pcb/uwu/amount/derived/termodynamics/PascalsTest.java
package pcb.uwu.amount.derived.termodynamics;

import org.junit.Test;
import pcb.uwu.amount.derived.fundamental.Area;
import pcb.uwu.amount.derived.fundamental.Volume;
import pcb.uwu.amount.derived.mechanics.Newtons;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;
import static pcb.uwu.unit.derived.area.SquareMeterUnit.SQUAREMETER;
import static pcb.uwu.unit.derived.fundamental.CubicMeterUnit.CUBIC_METER;

public class PascalsTest {
	private static final Pascals pascals = new Pascals(9);

	@Test
	public void multipliedByVolume() {
		assertEquals(new Joules(27),
				pascals.multipliedBy(new Volume(3, CUBIC_METER), DECIMAL64));
	}

	@Test
	public void multipliedByArea() {
		assertEquals(new Newtons(18),
				pascals.multipliedBy(new Area(2, SQUAREMETER), DECIMAL64));
	}
}

File: implementation/src/test/java/pcb/uwu/amount/derived/thermodynamics/PascalsTest.kt
package pcb.uwu.amount.derived.thermodynamics

import pcb.uwu.amount.derived.geometry.area.SquareMeters
import pcb.uwu.amount.derived.geometry.volume.CubicMeters
import pcb.uwu.amount.derived.mechanics.Newtons
import kotlin.test.Test
import kotlin.test.assertEquals

class PascalsTest
{
    @Test
    fun multipliedByVolume() =
        assertEquals(Joules(27),
                     Pascals(9) * CubicMeters(3))

    @Test
    fun multipliedByArea() =
        assertEquals(Newtons(18),
                     Pascals(9) * SquareMeters(2))
}
